src/guardian/staking-module-guard/staking-module-guard.service.ts

```diff
@@ -156,6 +227,43 @@ export class StakingModuleGuardService {
     return attackIntersections;
   }
 
+  public async getIntersectionBetweenUsedAndUnusedKeys(
+    intersectionsWithLidoWC: VerifiedDepositEvent[],
+    blockData: BlockData,
+  ) {
+    const depositedPubkeys = intersectionsWithLidoWC.map(
+      (deposit) => deposit.pubkey,
+    );
+
+    if (!depositedPubkeys.length) {
+      return [];
+    }
+
+    this.logger.log(
+      'Found intersections with lido credentials, need to check duplicated keys',
+    );
+
+    const { data, meta } =
+      await this.stakingRouterService.getKeysWithDuplicates(depositedPubkeys);
+
+    if (meta.elBlockSnapshot.blockNumber < blockData.blockNumber) {
```

Comment:
It would be good to add a `blockHash` check as was done earlier

```js
  public isNeedToProcessNewState(newMeta: {
    blockHash: string;
    blockNumber: number;
  }) {
    const lastMeta = this.lastProcessedStateMeta;
    if (!lastMeta) return true;
    if (lastMeta.blockNumber > newMeta.blockNumber) {
      this.logger.error('Keys-api returns old state', newMeta);
      return false;
    }
    return lastMeta.blockHash !== newMeta.blockHash;
  }
```

Reply:
i think code will be easier to read if we use oldMeta from method parameter
i dont see here a reason for cache this value

Reply:
but for readability i will add function for condition

Reply:
as we know that used keys shoudnt be deleted

Reply:
Is the blockHash check not needed here?